### PR TITLE
Incorporate PSScriptAnalyzerSettings for Enhanced Code Quality 

### PR DESCRIPTION
**Pull request by mdaneri**
The purpose of this improvement request is to propose the inclusion of PSScriptAnalyzerSettings.psd1 in the Pode project. PowerShell Script Analyzer is a static code checker for PowerShell scripts and modules that encourages best practices and improves code quality. By integrating PSScriptAnalyzerSettings.psd1, we can define custom rules and configurations specific to Pode's needs, ensuring consistent code style, adherence to best practices, and early detection of potential issues.

Issue #1244

### CODE REVIEW

File: PSScriptAnalyzerSettings.psd1 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# PSScriptAnalyzerSettings.psd1
+@{
+    Severity     = @('Error', 'Warning', 'Information')
+    ExcludeRules = @('PSAvoidUsingCmdletAliases' ,'PSAvoidUsingPlainTextForPassword','PSAvoidUsingWriteHost','PSAvoidUsingInvokeExpression','PSUseShouldProcessForStateChangingFunctions',
+    'PSAvoidUsingUsernameAndPasswordParams','PSUseProcessBlockForPipelineCommand','PSAvoidUsingConvertToSecureStringWithPlainText')
+}
```
Comment (Badgerati) [inline PSScriptAnalyzerSettings.psd1:6]: Could we add `PSUseSingularNouns` to the excludes list, as there are valid uses of plural functions :)

**Review by Badgerati — CHANGES_REQUESTED** [review]

File: PSScriptAnalyzerSettings.psd1 (hunk 1 above)
Comment (mdaneri) [reply PSScriptAnalyzerSettings.psd1:6]: Sure, but I'm going to open an improvement request to rename all internal function to singular name :)

**Review by Badgerati — APPROVED** [review]

**Comment by Badgerati** [comment]
I spotted in #1244 you reference updating the docs, is this something you want to do for the PR or good to merge?

**Comment by mdaneri** [comment]
I think it's good to go
renaming the internal functions is something for a future version

**Pull request merged**